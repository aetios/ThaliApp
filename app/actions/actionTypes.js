export const NAVIGATE = 'NAVIGATE';
export const BACK = 'BACK';
export const OPENDRAWER = 'OPENDRAWER';
export const CALENDARRETREIVED = 'CALENDARRETREIVED';
export const CALENDARERROR = 'CALENDARERROR';
export const RESETLOGINSTATE = 'RESETLOGINSTATE';
export const WELCOME = 'WELCOME';
