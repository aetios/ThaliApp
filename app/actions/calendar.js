/**
 * Created by aucke on 3/8/17.
 */
