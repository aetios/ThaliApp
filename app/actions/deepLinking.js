export const DEEPLINK = 'DEEPLINKING_DEEPLINK';

export function deepLink(url, stayInApp = true) {
  return {
    type: DEEPLINK,
    payload: { url, stayInApp },
  };
}
