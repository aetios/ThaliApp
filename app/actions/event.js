export const EVENT = 'EVENT_EVENT';
export const FETCHING = 'EVENT_FETCHING';
export const SUCCESS = 'EVENT_SUCCESS';
export const FAILURE = 'EVENT_FAILURE';

export function event(pk, token) {
  return {
    type: EVENT,
    payload: { pk, token },
  };
}

export function success(eventData, eventRegistrations) {
  return {
    type: SUCCESS,
    payload: { eventData, eventRegistrations },
  };
}

export function fetching() {
  return {
    type: FETCHING,
  };
}

export function failure() {
  return {
    type: FAILURE,
  };
}
