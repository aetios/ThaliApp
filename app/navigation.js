import {
  createDrawerNavigator,
  createStackNavigator,
  createSwitchNavigator,
  NavigationActions,
  DrawerActions,
} from 'react-navigation';

import Login from './ui/screens/login/LoginScreenContainer';
import Welcome from './ui/screens/welcome/WelcomeScreenContainer';
import Event from './ui/screens/events/EventScreenContainer';
import Calendar from './ui/screens/events/CalendarScreenContainer';
import Profile from './ui/screens/profile/ProfileScreenContainer';
import Pizza from './ui/screens/pizza/PizzaScreenContainer';
import Registration from './ui/screens/events/RegistrationScreenContainer';
import MemberList from './ui/screens/memberList/MemberListScreenContainer';
import SplashScreen from './ui/screens/splash/SplashScreen';
import Settings from './ui/screens/settings/SettingsScreenContainer';
import EventAdmin from './ui/screens/events/EventAdminScreenContainer';
import Sidebar from './ui/components/sidebar/SidebarContainer';

const MainNavigator = createDrawerNavigator({
  Welcome,
  Calendar,
  MemberList,
  Settings,
}, {
  contentComponent: Sidebar,
});

const SignedInNavigator = createStackNavigator({
  MainNavigator,
  Event,
  Profile,
  Pizza,
  Registration,
  EventAdmin,
}, {
  headerMode: 'none',
});

const AppNavigator = createSwitchNavigator({
  Splash: SplashScreen,
  Auth: Login,
  SignedIn: SignedInNavigator,
});

let navigator;

function setTopLevelNavigator(navigatorRef) {
  navigator = navigatorRef;
}

function navigate(routeName, params) {
  navigator.dispatch(
    NavigationActions.navigate({
      routeName,
      params,
    }),
  );
}

function goBack() {
  navigator.dispatch(
    NavigationActions.back(),
  );
}

function toggleDrawer() {
  navigator.dispatch(
    DrawerActions.toggleDrawer(),
  );
}

export default {
  AppNavigator,
  navigate,
  goBack,
  toggleDrawer,
  setTopLevelNavigator,
};
