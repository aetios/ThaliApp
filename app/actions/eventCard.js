import * as types from "./actionTypes";
export function openEvent(eventId) {
  console.log("opening event");
  return {
  };
}
