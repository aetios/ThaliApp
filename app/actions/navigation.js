import * as types from './actionTypes';

export function navigate(scene, newSection = false) {
  return {
    type: types.NAVIGATE,
    scene,
    newSection,
  };
}

export function back() {
  return {
    type: types.BACK,
  };
}

export function updateDrawer(drawerState) {
  return {
    type: types.OPENDRAWER,
    drawerOpen: drawerState,
  };
}
