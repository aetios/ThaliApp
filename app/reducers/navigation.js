import * as navigationActions from '../actions/navigation';
import * as sessionActions from '../actions/session';
import { LOGIN_SCENE, SPLASH_SCENE } from '../ui/components/navigator/scenes';

const initialState = {
  previousScenes: [],
  currentScene: SPLASH_SCENE,
  loggedIn: false,
  drawerOpen: false,
};


export default function navigate(state = initialState, action = {}) {
  const { currentScene, previousScenes, drawerOpen } = state;
  switch (action.type) {
    case sessionActions.SIGNED_IN: {
      return {
        ...state,
        loggedIn: true,
      };
    }
    case navigationActions.BACK: {
      if (drawerOpen) {
        return {
          ...state,
          drawerOpen: false,
        };
      }
      const scene = previousScenes.pop();
      return {
        ...state,
        previousScenes,
        currentScene: scene,
      };
    }
    case navigationActions.NAVIGATE: {
      if (action.payload.scene === currentScene) {
        return {
          ...state,
          drawerOpen: false,
        };
      } if (action.payload.newSection) {
        return {
          ...state,
          previousScenes: [],
          currentScene: action.payload.scene,
          drawerOpen: false,
        };
      }
      return {
        ...state,
        previousScenes: [
          ...previousScenes,
          currentScene,
        ],
        currentScene: action.payload.scene,
        drawerOpen: false,
      };
    }
    case navigationActions.OPENDRAWER: {
      return {
        ...state,
        drawerOpen: action.payload.drawerOpen,
      };
    }
    case sessionActions.TOKEN_INVALID:
    case sessionActions.SIGN_OUT: {
      return {
        ...initialState,
        currentScene: LOGIN_SCENE,
      };
    }
    default:
      return state;
  }
}
