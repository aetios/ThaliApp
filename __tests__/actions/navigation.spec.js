import * as actions from '../../app/actions/navigation';

describe('navigation actions', () => {
  it('should expose the navigation actions', () => {
    expect(actions.NAVIGATE).toEqual('NAVIGATE_NAVIGATE');
    expect(actions.BACK).toEqual('NAVIGATE_BACK');
    expect(actions.OPENDRAWER).toEqual('NAVIGATE_OPENDRAWER');
  });

  it('should create an action to navigate to a screen', () => {
    expect(actions.navigate('scene')).toMatchSnapshot();
    expect(actions.navigate('scene', true)).toMatchSnapshot();
  });

  it('should create an action to navigate back', () => {
    expect(actions.back()).toMatchSnapshot();
  });

  it('should create an action to update the drawer state', () => {
    expect(actions.updateDrawer('state')).toMatchSnapshot();
  });
});