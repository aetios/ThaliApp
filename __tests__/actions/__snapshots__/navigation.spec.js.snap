// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navigation actions should create an action to navigate back 1`] = `
Object {
  "type": "NAVIGATE_BACK",
}
`;

exports[`navigation actions should create an action to navigate to a screen 1`] = `
Object {
  "payload": Object {
    "newSection": false,
    "scene": "scene",
  },
  "type": "NAVIGATE_NAVIGATE",
}
`;

exports[`navigation actions should create an action to navigate to a screen 2`] = `
Object {
  "payload": Object {
    "newSection": true,
    "scene": "scene",
  },
  "type": "NAVIGATE_NAVIGATE",
}
`;

exports[`navigation actions should create an action to update the drawer state 1`] = `
Object {
  "payload": Object {
    "drawerOpen": "state",
  },
  "type": "NAVIGATE_OPENDRAWER",
}
`;
